src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/LanguageServerTarget.cs

```diff
@@ -0,0 +1,427 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.Contracts;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using StreamJsonRpc;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface ILspServiceProvider
+{
+    ILspServices CreateServices(string serverKind);
```

Comment:
This feels a little Roslyny. Do we need this in CLaSP?

Reply:
Hmm, I'll try remove it and see how it goes.

Reply:
You make a good point @NTaylorMullen, it was pretty Roslyn-targeted. I think the solution here might be my favorite solution: Less code. If we do away with ILspServiceProvider completely and just let the LanguageServerTarget construct its own in the black box of `GetLspServiceProvider` (soon to be `GetLspServices`, dono why I gave it a confusing name) then we can avoid having opinions on what pieces are "needed" to construct it.

Reply:
@dibarbet I'd be curious to hear if you think moving to CLaSP would help or hinder a move off of MEF, and if you notice any tweaks which could make it more "help". I'm hoping that the abstractions here could help push away from implementation details touching every part of the LSP as I've seen happen a bit in both Razor and Roslyn.

Reply:
Not sure.  We could definitely leave it up to the users of the library to implement the service provider.  But that is also one of the more interesting pieces that it seems like the library should automatically handle (creation of new services when servers restart / per server instances).

We definitely need to provide a sane default that utilizes some DI library.

Reply:
While I agree that we should offer a default I don't see it as a top priority since both of our first 2 customers (Roslyn and Razor) already have their own system.

Since y'all are not happy with your DI setup I would suggest that using the DI system that Razor uses (forget the name ATM) is a good default.

Reply:
Maybe we could utilize [Microsoft.Extensiosn.DependencyInjection.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection.Abstractions)? This provides service containers and there are hooks in the ecosystem to allow MEF/non-MEF bits to hook in?